packages/fuels-programs/src/contract.rs

```diff
@@ -175,17 +167,46 @@ impl DeployConfiguration {
 /// compiling, deploying, and running transactions against a contract.
 /// The contract has a wallet attribute, used to pay for transactions and sign them.
 /// It allows doing calls without passing a wallet/signer each time.
-pub struct Contract;
+pub struct Contract {
+    binary: Vec<u8>,
+    salt: Salt,
+    storage_slots: Vec<StorageSlot>,
+    contract_id: ContractId,
+    state_root: Bytes32,
+}
 
 impl Contract {
-    pub fn compute_contract_id_and_state_root(
-        compiled_contract: &CompiledContract,
+    pub fn new(binary: Vec<u8>, salt: Salt, storage_slots: Vec<StorageSlot>) -> Self {
```

Comment:
An option would be to make `deploy_loaded` public. Then we could do `Contract::new(...).deploy_loaded(...)`

Reply:
Can we please do this in another PR? This PR is supposed to be here just for the `get_contract_id` UX.

Reply:
The UX for `get_contract_id ` already leads to a breaking change. I'm leaning towards resolving such things immediately instead of accumulating breaking changes to the same part of our UI in the pipeline.